src/parsers/manifest/dash/parse_mpd.ts

```diff
@@ -277,5 +275,34 @@ function parseCompleteIntermediateRepresentation(
     }
   }
 
+  let duration: number|undefined;
+
+  const { minimumTime, maximumTime } = parsedMPD;
+  if (maximumTime?.isContinuous === false &&
+      minimumTime?.isContinuous === false) {
+    duration = (maximumTime.value - minimumTime.value);
+  } else {
+    if (rootAttributes.duration !== undefined) {
+      const firstPosition = parsedPeriods.reduce((acc: number|undefined, { start }) => {
+        return acc !== undefined ? Math.min(acc, start) : start;
+      }, undefined);
```

Comment:
Still not sure about that one. Why not take minimumTime if it exists?

Reply:
I assume there is a coherence between a pair of start/end values. Whether they come from manifest or indexes, i don't know if it is a good idea to try to mix up pairs.